### PR TITLE
Task2698 material UI tests implement inputs text field

### PR DESCRIPTION
**Pull request by nghiatt1509**
Implement TextFields Test

### CODE REVIEW

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/inputs/TextField.java [hunk 1]
```diff
@@ -41,4 +69,8 @@ public TextFieldAssert has() {
         return this.is();
     }
 
+    @JDIAction("Click to focus on '{name}' textfield")
+    public void click() {
+        core().click();
```
Comment (cold-flame) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/inputs/TextField.java:74]: mb prefer to use smth like this **new Actions(driver).moveToElement(element).click().perform();**

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/TextFieldTests.java [hunk 2]
```diff
@@ -1,63 +1,194 @@
...
+        Assert.assertEquals(textFieldPage.selectElementField.getText(), CurrencyItems.EUR.currencyItems);
+        textFieldPage.selectElementField.click();
+        textFieldPage.selectElement.selectItemByText(CurrencyItems.BTC.currencyItems);
+        Assert.assertEquals(textFieldPage.selectElementField.getText(), CurrencyItems.BTC.currencyItems);
+        textFieldPage.selectElementField.click();
+        textFieldPage.selectElement.selectItemByText(CurrencyItems.JPY.currencyItems);
+        Assert.assertEquals(textFieldPage.selectElementField.getText(), CurrencyItems.JPY.currencyItems);
+
+        textFieldPage.selectNativeSelect.select(CurrencyItems.USD.currencyItems);
+        Assert.assertEquals(textFieldPage.selectNativeSelect.getText(), CurrencyItems.USD.currencyItems);
```
Comment (cold-flame) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/TextFieldTests.java:159]: just try to use JDI Assert with **has()** action

Comment (tsvetaevAlex) [reply jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/TextFieldTests.java:159]: try to reduce lines of code using foreach loop using enum you created: enum CurrencyItems

File: jdi-light/src/main/java/com/epam/jdi/light/elements/common/UIElement.java [hunk 3]
```diff
@@ -147,7 +149,21 @@ public void sendKeys(CharSequence... value) {
...
+        String stringBeforeClear = get().getText();
+        get().clear();
+        String stringAfterClear = get().getText();
+
+        if(stringBeforeClear.equals(stringAfterClear)) {
+            get().click();
+            if (getOs() == WIN){
+                get().sendKeys(Keys.CONTROL + "A");
+                get().sendKeys(Keys.BACK_SPACE);
+            } else {
```
[the remark this answers is not in the document]
Comment (nghiatt1509) [reply jdi-light/src/main/java/com/epam/jdi/light/elements/common/UIElement.java:162]: I didn't have environment for Linux so I can not verify this on Linus.

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/TextFieldTests.java (hunk 2 above)
Comment (nghiatt1509) [reply jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/TextFieldTests.java:159]: Refactored!

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/inputs/TextField.java (hunk 1 above)
Comment (nghiatt1509) [reply jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/inputs/TextField.java:74]: I see click() is defined from UI.Element class and our page class is inherited from it. So I think we should reuse this click().

Comment (cold-flame) [reply jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/inputs/TextField.java:67]: ok

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/TextFieldTests.java [hunk 4]
```diff
@@ -1,63 +1,161 @@
...
-        textFieldFrame.textField.setValue("17:96");
-        textFieldFrame.textField.has().text("17:09");
-        textFieldFrame.textField.setValue("17:75");
-        textFieldFrame.textField.has().text("17:07");
-        textFieldFrame.textField.setValue("17:x5");
-        textFieldFrame.textField.has().text("17:05");
-    }
+    public void selectTest() {
 
+        for(CurrencyItems currency : CurrencyItems.values()){
```
Comment (cold-flame) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/TextFieldTests.java:152]: excellent decision !

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/TextFieldTests.java [hunk 5]
```diff
@@ -1,63 +1,161 @@
...
 
 public class TextFieldTests extends TestsInit {
 
-    public static final String DEFAULT_VALUE = "Default value";
+    public static final String DEFAULT_VALUE = "Default Value";
+
+    @BeforeMethod
+    public void openTextFieldsPage(){
+        textFieldPage.open();
+        jdiAssert(textFieldPage.isOnPage(textFieldPage.url), Matchers.is(true));
```
Comment (cold-flame) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/TextFieldTests.java:24]: use textFieldPage.isOpened()

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/TextFieldTests.java [hunk 6]
```diff
@@ -1,63 +1,161 @@
 package io.github.epam.material.tests.inputs;
 
+import com.epam.jdi.light.material.elements.utils.enums.CurrencyItems;
+import io.github.com.StaticSite;
 import io.github.epam.TestsInit;
+import org.hamcrest.Matchers;
+import org.testng.Assert;
```
Comment (cold-flame) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/inputs/TextFieldTests.java:7]: delete unnecessary imports

**Review by cold-flame — CHANGES_REQUESTED** [review]
Need to fix some code

**Review by tsvetaevAlex — APPROVED** [review]

**Review by cold-flame — APPROVED** [review]

**Pull request merged**